### PR TITLE
✨(newsletter) store user newsletter subscription

### PR DESCRIPTION
**Pull request by kernicPanel**
We need to store newsletter subscription preference in Joanie. A checkbox is added to the registration form, and related value is stored in a cookie.
Fonzie will use that cookie to add the subscription to the user token payload which will be sent to Joanie, updating the user.

Related PRs : https://github.com/openfun/fonzie/pull/36 https://github.com/openfun/joanie/pull/701

### CODE REVIEW

**Review by jonathanreveille — APPROVED** [review]

File: funsite/templates/lms/register.html [hunk 1]
```diff
@@ -152,6 +154,12 @@ <h2 class="color-regal-blue">
                             link_start='<a href="{url}"  tabindex="-1">'.format(url=honor_code_path),
                             link_end='</a>')}</label>
                     </div>
+
+                    <div class="form-group" id="field-newsletter">
+                        <input id="newsletter" type="checkbox" value="true" />
+                        <label for="newsletter">${_('I would like to receive updates from {platform_name}').format(
```
Comment (lunika) [inline funsite/templates/lms/register.html]: You also have to add the translation in your PR, they are not managed in another service. You will found them in `funsite/locale`

And the exact expected translation is 
```
J'accepte de recevoir vos emails sur l’offre de formations professionnelles payantes FUN et confirme avoir pris connaissance des [Conditions générales de la newsletter formation professionnelle](https://www.fun-mooc.fr/fr/cgu/cgu-newsletter-formapro/).
```

That also means a link is present in the sentence.

File: funsite/locale/fr/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -294,6 +294,10 @@ msgstr "Inscrivez-vous sur FUN !"
 msgid "Sign up to {platform_name}"
 msgstr "Inscription à {platform_name}"
 
+#: /edx/app/edxapp/fun-apps/funsite/templates/lms/register.html:0
+msgid "I agree to receive your emails on the offer of paid professional training FUN and confirm that I have read the <a href='https://www.fun-mooc.fr/fr/cgu/cgu-newsletter-formapro/' tabindex='-1'>General conditions of the professional training newsletter</a>"
```
Comment (jonathanreveille) [inline funsite/locale/fr/LC_MESSAGES/django.po]:
```suggestion
msgid "I agree to receive your emails regarding the offer of paid professional training from FUN and confirm that I have read the <a href='https://www.fun-mooc.fr/fr/cgu/cgu-newsletter-formapro/' tabindex='-1'>General Conditions of the Professional Training Newsletter</a>."
```
WDYT?

**Review by jonathanreveille — APPROVED** [review]

File: funsite/locale/fr/LC_MESSAGES/django.po (hunk 2 above)
Comment (lunika) [inline funsite/locale/fr/LC_MESSAGES/django.po]:
```suggestion
msgid "I agree to receive your emails on the offer of paid professional training FUN and confirm that I have read the <a href='https://www.fun-mooc.fr/en/cgu/cgu-newsletter-formapro/' tabindex='-1'>General conditions of the professional training newsletter</a>"
```

File: funsite/templates/lms/register.html [hunk 3]
```diff
@@ -159,6 +161,11 @@ <h2 class="color-regal-blue">
                             link_start='<a href="{url}"  tabindex="-1">'.format(url=honor_code_path),
                             link_end='</a>')}</label>
                     </div>
+
+                    <div class="form-group" id="field-newsletter">
+                        <input id="newsletter" type="checkbox" value="true" />
+                        <label for="newsletter">${_("I agree to receive your emails on the offer of paid professional training FUN and confirm that I have read the <a href='https://www.fun-mooc.fr/fr/cgu/cgu-newsletter-formapro/' tabindex='-1'>General conditions of the professional training newsletter</a>")}</label>
```
Comment (lunika) [inline funsite/templates/lms/register.html]:
```suggestion
                        <label for="newsletter">${_("I agree to receive your emails on the offer of paid professional training FUN and confirm that I have read the <a href='https://www.fun-mooc.fr/en/cgu/cgu-newsletter-formapro/' tabindex='-1'>General conditions of the professional training newsletter</a>")}</label>
```

File: funsite/templates/lms/register.html [hunk 4]
```diff
@@ -16,6 +16,8 @@
      (function() {
          $('.register-form').on('submit', function(event) {
              event.preventDefault();
+             document.cookie = 'newsletter_subscribed=' + $('#newsletter').is(':checked');
```
Comment (jbpenrath) [inline funsite/templates/lms/register.html:19]: jQuery 🫶

Just to let you know the JS alternative
```js
document.getElementById('newsletter').checked
```

File: funsite/templates/lms/register.html [hunk 5]
```diff
@@ -159,6 +161,12 @@ <h2 class="color-regal-blue">
                             link_start='<a href="{url}"  tabindex="-1">'.format(url=honor_code_path),
                             link_end='</a>')}</label>
                     </div>
+
+                    <div class="form-group" id="field-newsletter">
+                        <input id="newsletter" type="checkbox" value="true" />
```
Comment (jbpenrath) [inline funsite/templates/lms/register.html]: I do not understand the purpose of `value=true`
```suggestion
                        <input id="newsletter" type="checkbox" />
```

**Review by lunika — APPROVED** [review]

**Review by jonathanreveille — APPROVED** [review]

**Pull request merged**